jax/experimental/sparse/nm.py

```diff
@@ -181,6 +181,9 @@ def _nm_spmm_abstract_eval(
 if gpu_sparse.cuda_is_supported:
   mlir.register_lowering(nm_spmm_p, _nm_spmm_gpu_lowering, platform="cuda")
 
+if gpu_sparse.rocm_is_supported:
+  mlir.register_lowering(nm_spmm_p, _nm_spmm_gpu_lowering, platform="ROCM")
```

Comment:
Nit: elsewhere we're using "rocm" (lower case), probably best to be consistent.

Reply:
@hawkinsp Whenever we use `platform="ROCM"`. I observed we used caps. But, I can fix all of those if you would like

Reply:
Hmm. Interesting. No, I'd stick with "ROCM" then. I don't see any logic to canonicalize the platform name.

Reply:
> Hmm. Interesting. No, I'd stick with "ROCM" then. I don't see any logic to canonicalize the platform name.

I also observed discrepancies in cuda. i.e. sometimes `platform="CUDA"` while other times platform="cuda"